dev/domain-controller/new-admin-vm/Install-CAM.ps1

```diff
@@ -1320,7 +1330,159 @@ brokerLocale=en_US
+				{
+					return $true
+				} else {
+					return $false
+				}
+			}
+
+            SetScript  = {
+				##
+				# Do this so SSL Errors are ignored
+				add-type @"
+	using System.Net;
+	using System.Security.Cryptography.X509Certificates;
+	public class TrustAllCertsPolicy : ICertificatePolicy {
+		public bool CheckValidationResult(
+			ServicePoint srvPoint, X509Certificate certificate,
+			WebRequest request, int certificateProblem) {
+			return true;
+		}
+	}
+"@
+				[System.Net.ServicePointManager]::CertificatePolicy = New-Object TrustAllCertsPolicy
+				##
+
+				# Read in Authorization Information
+				Get-Content "$env:AZURE_AUTH_LOCATION" | Foreach-Object{
+					$var = $_.Split('=', 2)
+					New-Variable -Name $var[0] -Value $var[1]
+				}
+
+				$camSaasBaseUri = $using:camSaasUri
+				$camRegistrationError = ""
+				for($idx = 0; $idx -lt $using:retryCount; $idx++) {
+					try {
+						$userRequest = @{
+							username = $client
+							password = $key
+							tenantId = $tenant
+						}
+						$registerUserResult = ""
+						try {
+							$registerUserResult = Invoke-RestMethod -Method Post -Uri ($camSaasBaseUri + "/api/v1/auth/users") -Body $userRequest
+						} catch {
+							$registerUserResult = ConvertFrom-Json $_.ErrorDetails.Message
+						}
+						Write-Verbose (ConvertTo-Json $registerUserResult)
+						# Check if registration succeeded or if it has been registered previously
+						if( !(($registerUserResult.code -eq 201) -or ($registerUserResult.data.reason.ToLower().Contains("already exist"))) ) {
+							throw ("Failed to register with CAM. Result was: " + (ConvertTo-Json $registerUserResult))
+						}
+
+						[System.Environment]::SetEnvironmentVariable("CAM_USERNAME", $userRequest.username, "Machine")
+						[System.Environment]::SetEnvironmentVariable("CAM_PASSWORD", $userRequest.password, "Machine")
+						[System.Environment]::SetEnvironmentVariable("CAM_TENANTID", $userRequest.tenantId, "Machine")
+						[System.Environment]::SetEnvironmentVariable("CAM_URI", $camSaasBaseUri, "Machine")
+						Write-Host "Cloud Access Manager PoC has been registered succesfully"
+
+						# Get a Sign-in token
+						$signInResult = ""
+						try {
+							$signInResult = Invoke-RestMethod -Method Post -Uri ($camSaasBaseUri + "/api/v1/auth/signin") -Body $userRequest
+						} catch {
+							$signInResult = ConvertFrom-Json $_.ErrorDetails.Message
+						}
+						Write-Verbose ((ConvertTo-Json $signInResult) -replace "\.*token.*", 'Token": "Sanitized"')
+						# Check if signIn succeded
+						if ($signInResult.code -ne 200) {
+							throw ("Signing in failed. Result was: " + (ConvertTo-Json $signInResult))
+						}
+						$tokenHeader = @{
+							authorization=$signInResult.data.token
+						}
+						Write-Host "Cloud Access Manager sign in succeeded"
+
+						$rcCred = $using:registrationCodeAsCred
+						$registrationCode = $rcCred.Password
```

Comment:
Does that work? I thought you needed to 'getnetworkcredentials.password' or something like that otherwise the blob stays encrypted.